pkg/logql/syntax/linefilter.go

```diff
@@ -0,0 +1,73 @@
+package syntax
+
+import (
+	"encoding/binary"
+
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+// Binary encoding of the LineFilter
+// integer is varint encoded
+// strings are variable-length encoded
+//
+// +---------+--------------+-------------+
+// | Ty      | Match        | Op          |
+// +---------+--------------+-------------+
+// | value   | len  | value | len | value |
+// +---------+--------------+-------------+
```

Comment:
That's probably a nice compromise but I'd go with the JSON one we have to keep consistency.

Reply:
Consistent with the Plan serialization?

Reply:
I used the encoding utils that we already have, which makes the code much simpler: https://github.com/grafana/loki/pull/11492/commits/335d79ce4a909bc2738b52cbf2934c76c5aa1a63

Reply:
Nice. Something to consider for the query plan.